wallet/wallet.c

```diff
@@ -2139,7 +2139,8 @@ struct outpoint *wallet_outpoint_for_scid(struct wallet *w, tal_t *ctx,
 			  "FROM utxoset "
 			  "WHERE blockheight = ?"
 			  " AND txindex = ?"
-			  " AND outnum = ?");
+			  " AND outnum = ?"
+			  " AND spendheight IS NOT NULL");
```

Comment:
Isn't this inverted? We're interested in the ones that aren't spent yet.

Reply:
That would explain the test breakage :)